### ISSUE TITLE
Check package versions before installing

### ISSUE THREAD

**Issue by ashiklom** [r-lib/remotes#76]
Suppose I have `mypackage` with the following in its `DESCRIPTION` file:
```
Imports:
  otherpackage (>=1.0.0)
Remotes:
  github::username/otherpackage
```
Suppose also that I am developing `otherpackage` locally, side-by-side with `mypackage`, and that my _local_ installed version of `otherpackage` is 1.0.0.9000. 

If I just run `devtools::install("mypackage")` in this situation, it will install the _older_ version of `otherpackage` from GitHub, rather than recognizing that a newer version that already satisfies the dependency is available locally. This seems like the wrong behavior...?

**Comment by gaborcsardi** [comment]
Probably. I often want the tip of the branch, though.

**Comment by ashiklom** [comment]
I disagree. I can't imagine a situation where a package will have a lower version in its DESCRIPTION but be at a later point in the Git history (short of a mistake, or bad package/git management). On the other hand, I can think of many situations where the package version on Git(Hub) will be lower than the currently installed one, and in those cases, I don't think the remote install should happen.

Basically, I think remotes should follow the following logic:
1. Get installed package version `v_install`
2. Get remote package version `v_remote`
3. `if (v_remote > v_install) install_remote(...) else pass`

(PS. Sorry for the close and re-open -- accidentally hit the "close" button!)

**Comment by gaborcsardi** [comment]
> I can't imagine a situation where a package will have a lower version in its DESCRIPTION but be at a later point in the Git history

It is not lower, it is the same.

**Comment by gaborcsardi** [comment]
Ideally there should be also a way to say that you want a certain commit or anything newer.

**Comment by ashiklom** [comment]
> It is not lower, it is the same. Very often people do not increase the version number with each commit.

I guess I wasn't clear in my original issue. If this only happened when the versions were the same, that would be understandable. However, I'm finding that the dependency is reinstalled even when my local package is a newer version than what's on GitHub; the only time the install is skipped is if the packages are identical. I agree that package versions may not necessarily be updated with every commit, but it would be nice to have the option of preventing package downgrades via package versions.

**Comment by gaborcsardi** [comment]
This will not work, unfortunately. If you specify `repo/package` in `Remotes`, then remotes will make  sure that you have the tip of the branch, even if you have sg newer locally. We don't really have a way to tell that you have something newer.
